### PR TITLE
Add support for installation via pip

### PR DESCRIPTION
**Pull request by lonvia**
Now that Nominatim can be run as a pure Python application, it is possible to support installation via a simple `pip install`. This PR adds support for that.

It is not possible to use the 'nominatim' namespace for a public Python package because it is [already taken](https://pypi.org/project/nominatim/) and the author is not reachable. I've therefore split up the code into 'nominatim-db' (the part for importing and maintaining the database) and 'nominatim-api' (the frontend library and server code). This turned out to be a quite natural division. The only shared part of the code are the configuration files and the code that reads them. These parts are duplicated in each package.

The code is organised in a way that it is possible to run nominatim directly from the source tree, if required. This ensures that the setup for development remains a simple one.

The biggest drawback of the split-up is that naming and the position has changed for every single Python file and every import. If you are running the Python frontend, you need to change the server module from `nominatim.server.falcon.server` to `nominatim_api.server.falcon.server`. If you are using the Nominatim library, all imports need to be changed from `nominatim.api.<module>` to `nominatim_api.<module>`. If you have written custom tokenizers or sanitizers, the appropriate modules are now found in `nominatim_db`.

osm2pgsql will no longer be bundled with Nominatim when installing with pip. You need to get the package from your distribution. A minimum of 1.8 is required.

Installation via cmake (including the bundled osm2pgsql) remains possible throughout the 4.x series of Nominatim. 

The new Vagrant-compatible script for installation on Ubuntu 24.04 already uses the new pip-based installation method. The script for Ubuntu 22.04 will remain with the old cmake-based installation method because the osm2pgsql version that ships with Ubuntu 22.04 is too old. I've dropped support for Ubuntu 20.04.

While touching all files, I've also cleaned up the inconsistencies in the license notices. All Python code is now consistently licensed under GPL 3 and newer. The lua scripts are now available under Apache License, Version 2.0. And everything else remains under GPL 2 only.

Packages with this version are available as development packages on pypi.org. You can try the new packaging by doing a simple install via `pip install nominatim-db nominatim-api`. Keep in mind that this gets you a development version from master. Release versions will only be available with the next 4.5 release.

### CODE REVIEW

File: docs/admin/Migration.md [hunk 1]
```diff
@@ -15,6 +15,22 @@ breaking changes. **Please read them before running the migration.**
     If you are migrating from a version <3.6, then you still have to follow
     the manual migration steps up to 3.6.
 
+## 4.4.0 -> master
+
+### New structure for Python packages
+
+The nominatim Python package has been split into nominatim-db and nominatim-api.
```
Comment (mtmail) [inline docs/admin/Migration.md:22]: I'd use ` for the package names for readability

File: docs/develop/Development-Environment.md [hunk 2]
```diff
@@ -4,19 +4,19 @@ This chapter gives an overview how to set up Nominatim for development
 and how to run tests.
 
 !!! Important
-    This guide assumes that you develop under the latest version of Ubuntu. You
-    can of course also use your favourite distribution. You just might have to
-    adapt the commands below slightly, in particular the commands for installing
-    additional software.
+    This guide assumes that you develop under the latest version of Debain/Ubuntu.
```
Comment (mtmail) [inline docs/develop/Development-Environment.md:7]: 'that' can be omitted. Typo: Debian

File: docs/develop/Development-Environment.md [hunk 3]
```diff
@@ -85,24 +92,40 @@ composer global require "phpunit/phpunit=8.*"
...
+### Running Nominatim during development
+
+The source code for Nominatim can be found in the `src` directory and can
+be run in-place. The source directory features a special script
+`nominatim-cli.py` which does the same as the installed 'nominatim' binary
+but executes against the code in the source tree. For example:
+
+```
+me@machine:~$ cd Nomiantim
+me@machine:~Nomiantim$ ./nominatim-cli.py --version
```
Comment (mtmail) [inline docs/develop/Development-Environment.md:110]: Typo: Nominatim

File: docs/develop/Development-Environment.md [hunk 4]
```diff
@@ -85,24 +92,40 @@ composer global require "phpunit/phpunit=8.*"
...
+Nominatim version 4.4.99-1
+```
+
+Make sure you have activated the virtual environment that holds all
```
Comment (mtmail) [inline docs/develop/Development-Environment.md:114]: ... environment holding all ...

File: docs/develop/Development-Environment.md [hunk 5]
```diff
@@ -85,24 +92,40 @@ composer global require "phpunit/phpunit=8.*"
...
-To run all tests just go to the build directory and run make:
+To run all tests, run make from the source root:
 
 ```sh
-cd build
-make test
+make tests
 ```
 
+There are also goals for executing parts of the test suite: mypy, lint, pytest, bdd.
```
Comment (mtmail) [inline docs/develop/Development-Environment.md:127]: goal is the correct terms but might be unfamiliar. Maybe link to a documentation or be explicit ("make lint", "make bdd")

File: docs/library/Getting-Started.md [hunk 6]
```diff
@@ -12,22 +12,18 @@ in the database.
...
 Follow the [installation](../admin/Installation.md) and
 [import](../admin/Import.md) instructions to set up your database.
 
-It is not yet possible to install it in the usual way via pip or inside a
-virtualenv. To get access to the library you need to set an appropriate
-`PYTHONPATH`. With the default installation, the python library can be found
-under `/usr/local/share/nominatim/lib-python`. If you have installed
-Nominatim under a different prefix, adapt the `/usr/local/` part accordingly.
-You can also point the `PYTHONPATH` to the Nominatim source code.
+The Nominatim frontend library is contained in the 'nominatim-api' package.
```
Comment (mtmail) [inline docs/library/Getting-Started.md:21]: ` instead of '

File: packaging/nominatim-db/README.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+    pip install nominatim-db
+
+
+## Quick start
+
+First create a project directory for your new Nominatim database, which
+is the space for additional configuration and customization:
+
+    mkdir planet-project
```
Comment (mtmail) [inline packaging/nominatim-db/README.md:33]: Currently we use nominatim-project

File: vagrant/Install-on-Ubuntu-24.sh [hunk 8]
```diff
@@ -78,14 +67,14 @@ fi                                 #DOCS:
 # ---------------------
 #
 # Tune the postgresql configuration, which is located in 
-# `/etc/postgresql/12/main/postgresql.conf`. See section *Tuning the PostgreSQL database*
+# `/etc/postgresql/14/main/postgresql.conf`. See section *Tuning the PostgreSQL database*
```
Comment (mtmail) [inline vagrant/Install-on-Ubuntu-24.sh:70]: version 16

File: vagrant/Install-on-Ubuntu-24.sh [hunk 9]
```diff
@@ -122,32 +111,39 @@ if [ ! -f data/country_osm_grid.sql.gz ]; then       #DOCS:    :::sh
...
+    $USERHOME/nominatim-venv/bin/pip install packaging/nominatim-db
 
-    mkdir $USERHOME/build
-    cd $USERHOME/build
-    cmake $USERHOME/Nominatim
-    make
-    sudo make install
+# Nominatim is now ready to use. The nominatim binary is available at
+# `$USERHOME/venv/bin/nominatim`. If you want to have 'nominatim' in your
+# path, simply activate the virtual environment:
```
Comment (mtmail) [inline vagrant/Install-on-Ubuntu-24.sh:126]: . instead of :

File: packaging/nominatim-api/README.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+## Installation
+
+To install the Nominatim API from pypi, run:
+
+    pip install nominatim-api
+
+## Running a Nominatim server
+
+You need falcon or starlette to run Nominatim as a service, as well as
```
Comment (mtmail) [inline packaging/nominatim-api/README.md:18]: Falcon, Starlette

File: vagrant/Install-on-Ubuntu-24.sh [hunk 11]
```diff
@@ -122,32 +111,39 @@ if [ ! -f data/country_osm_grid.sql.gz ]; then       #DOCS:    :::sh
...
 # [importing a database from OSM data](../admin/Import.md). If you want to set up
 # the API frontend first, continue reading.
 #
 # Setting up the Python frontend
 # ==============================
 #
-# Some of the Python packages in Ubuntu are too old. Therefore run the
-# frontend from a Python virtualenv with current packages.
+# The Python frontend is contained in the nominatim-api package. To run
+# the API as a webservice, you also need falcon with uvicorn/gunicorn to
```
Comment (mtmail) [inline vagrant/Install-on-Ubuntu-24.sh:138]: The / might ambiguous, is that 'or' or 'in combination with' ?

File: packaging/nominatim-db/README.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+This module implements the database backend for Nominatim and the
+command-line tool for importing and maintaining the database.
+
+## Installation
+
+### Prerequisites
+
+Nominatim requires [osm2pgsql](https://osm2pgsql.org/) (>=1.8) for reading
+OSM data and [PostgreSQL](https://www.postgresql.org/) to store the data.
```
Comment (mtmail) [inline packaging/nominatim-db/README.md:15]: mention PostgreSQL minimal version here, too?

**Review by mtmail — COMMENTED** [review]
Amazing huge PR. I have mostly trivial wording suggestions.

**Comment by lonvia** [comment]
Thanks for reading through all of that. Should all be addressed now.

**Pull request merged**